core/server/data/schema/commands.js

```diff
@@ -145,6 +145,89 @@ async function dropUnique(tableName, columns, transaction) {
+            WHERE i.CONSTRAINT_TYPE = 'FOREIGN KEY'
+            AND i.CONSTRAINT_SCHEMA=:dbName
+            AND i.TABLE_NAME = :fromTable
+            AND k.COLUMN_NAME = :fromColumn
+            AND k.REFERENCED_TABLE_NAME = :toTable
+            AND k.REFERENCED_COLUMN_NAME = :toColumn`, {dbName, fromTable, fromColumn, toTable, toColumn});
+
+        return rawConstraints.length === 1;
+    } else {
+        const foreignKeys = await knex.raw(`PRAGMA foreign_key_list('${fromTable}');`);
+
+        const hasForeignKey = foreignKeys.some(foreignKey => foreignKey.table === toTable && foreignKey.from === fromColumn && foreignKey.to && toColumn);
+
+        return hasForeignKey;
+    }
+}
+
+/**
+ * Adds a foreign key to a table.
+ *
+ * @param {string} fromTableName - name of the table to add the foreign key to
+ * @param {string} fromColumn - column of the table to add the foreign key to
+ * @param {string} toTableName - name of the table to point the foreign key to
+ * @param {string} toColumn - column of the table to point the foreign key to
+ * @param {Object} transaction - connection object containing knex reference
+ * @param {Object} transaction.knex - knex instance
+ */
+async function addForeign(fromTable, fromColumn, toTable, toColumn, transaction) {
+    const hasForeignKey = await hasForeign(fromTable, fromColumn, toTable, toColumn, transaction);
+
+    if (!hasForeignKey) {
+        logging.info(`Adding foreign key for: ${fromColumn} in ${fromTable} to ${toColumn} in ${toTable}`);
+        return (transaction || db.knex).schema.table(fromTable, function (table) {
+            table.foreign(fromColumn).references(`${toTable}.${toColumn}`);
+        });
+    } else {
+        logging.warn(`Foreign key for ${fromColumn} in ${fromTable} to ${toColumn} in ${toTable} already exists`);
+    }
+}
+
+/**
+ * Drops a foreign key from a table.
+ *
+ * @param {string} fromTableName - name of the table to add the foreign key to
+ * @param {string} fromColumn - column of the table to add the foreign key to
+ * @param {string} toTableName - name of the table to point the foreign key to
+ * @param {string} toColumn - column of the table to point the foreign key to
+ * @param {Object} transaction - connection object containing knex reference
+ * @param {Object} transaction.knex - knex instance
+ */
+async function dropForeign(fromTable, fromColumn, toTable, toColumn, transaction) {
+    const hasForeignKey = await hasForeign(fromTable, fromColumn, toTable, toColumn, transaction);
+
+    if (hasForeignKey) {
+        logging.info(`Dropping foreign key for: ${fromColumn} in ${fromTable} to ${toColumn} in ${toTable}`);
+        return (transaction || db.knex).schema.table(fromTable, function (table) {
+            table.dropForeign(fromColumn);
+        });
+    } else {
+        logging.warn(`Dropping key for ${fromColumn} in ${fromTable} to ${toColumn} in ${toTable} does not exist`);
```

Comment:
```suggestion
        logging.warn(`Skipped dropping foreign key for ${fromColumn} in ${fromTable} to ${toColumn} in ${toTable} - foreign key does not exist`);
```

This would follow our general convention of `Skipped {something something} - {why?}` :wink: